avocado/core/job.py

```diff
@@ -89,7 +89,9 @@ def __init__(self, args=None):
         if unique_id is None:
             unique_id = job_id.create_unique_job_id()
         self.unique_id = unique_id
-        self.logdir = None
+        self._setup_job_results()
+        if not (self.standalone or getattr(self.args, "dry_run", False)):
+            self._update_latest_link()
```

Comment:
ACK to merging `_update_latest_link()` inside `_setup_job_results()`.

With regards to when we do the result init, we have to think about what we consider to be a completely ready Job.  So far, as described in the Job Phases RFC, the idea is that once instantiated, a Job is a full blown and capable entity.  Consider the following example:

```python
from avocado.core.job import Job
from avocado.core import exit_codes
from mylib import args, check_environment

job = Job(mydefaults)
job.log.info("Checking if environment is ready")
if not check_environment():
   job.log.error("Environment is not suitable for this job")
   job.exitcode = exit_codes.AVOCADO_JOB_FAIL
else:
   job.run()  # implies all job phases
sys.exit(job.exitcode) 
```

TBH, `job.log` logger does not *yet* exist, (a job logger reuses the test logger), but it really should.

Now, that succinct API usage is only possible because from the user standpoint, an instantiated Job, is a fully capable entity.  We could add obligations to the user, such as mandating that something like this is done:

```python

from avocado.core.job import Job
from avocado.core import exit_codes
from mylib import args, check_environment

job = Job(args)
job.setup_job_results() # or job.start()
...
```

But I don't think that improves the internal design of the job, and definetly the usability of the job class is much worse: users would have to know about this somewhat internal step.